src/connectors/openai.py

```
@@ -225,88 +225,100 @@ async def chat_completions(
-                # identity headers do not leak between calls. Downstream
-                # callers rely on identity-specific headers being scoped to
-                # a single request.
-                self.identity = identity
-                headers = self.get_headers()
-            except Exception:
-                headers = None
+            if not isinstance(request_data, ChatRequest):
+                raise TypeError(
+                    f"Expected ChatRequest or CanonicalChatRequest, got {type(request_data).__name__}. "
+                    "Backend connectors should only receive domain-format requests."
+                )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)